client/src/components/WorkflowInvocationState/WorkflowInvocationState.vue

```diff
@@ -107,14 +107,17 @@ const invocationStateSuccess = computed(() => {
 
 const workflowStore = useWorkflowStore();
 
-const isDeletedWorkflow = computed(() => getWorkflow()?.deleted === true);
-const workflowVersion = computed(() => getWorkflow()?.version);
+const workflow = ref<Workflow | undefined>(undefined);
+const isDeletedWorkflow = computed(() => workflow.value?.deleted === true);
+const workflowVersion = computed(() => workflow.value?.version);
 
 onMounted(async () => {
     try {
```

Comment:
I think you can pull all of this out of the onMounted hook ? Don't think we need to wait for the DOM to exist to start performing these calls.

Reply:
The reason I kept this all in the `onMounted` hook was because we had the 
```
await pollStepStatesUntilTerminal();
await pollJobStatesUntilTerminal();
```
calls in there that would start the timeout and keep trying to fetch even if there was an error raised from fetching the invocation. This way, we now only begin polling if we were able to fetch an invocation in the first place

Reply:
you can do this in a watcher.

Reply:
Doing this and fetching the invocation (the first time; and catching any error) in a watcher